types/header.go

```diff
@@ -7,8 +7,17 @@ import (
 	"time"
 
 	"github.com/celestiaorg/go-header"
+	tmbytes "github.com/tendermint/tendermint/libs/bytes"
 )
 
+type Hash = header.Hash
+
+func ConvertToHexBytes(hash Hash) tmbytes.HexBytes {
+	tmHash := make(tmbytes.HexBytes, 32)
```

Comment:
Why this has to be allocated instead of just using `return []byte(hash)`? Maybe it is something related to our tests? In tests data is not set properly (it's empty if we're not using it).

Reply:
I still think this has to be allocated because we have two different types so you have two allocate for both and the `free` memory space for the other one in this case. I know the underlying is still `[]byte`. Changing to what you suggested actually throws a segmentation fault.

Reply:
Both types have the same [underlying type](https://go.dev/ref/spec#Underlying_types) (`[]byte`), therefore they are [assignable](https://go.dev/ref/spec#Assignability), therefore they are [convertible](https://go.dev/ref/spec#Conversions).
Both types are [named](https://go.dev/ref/spec#Types), so the conversion has to be explicit (second rule in [assignability](https://go.dev/ref/spec#Assignability) chapter).